### ISSUE TITLE
unexpected status code 200 OK when using push_metrics

### ISSUE THREAD

**Issue by hvariant** [tikv/rust-prometheus#287]
I'm getting `Msg("unexpected status code 200 OK while pushing to ...")` when pushing to prometheus pushgateway docker [v0.10.0](https://hub.docker.com/layers/prom/pushgateway/latest/images/sha256-5e252ed423c397fcab03c718714464a0a8f8a0b51e075f38a38519a3eac8f171).

**To Reproduce**
Run the push example to send gauge metric to a pushgateway running in a docker container.

**Actual behavior**
Crash at `push_metrics(...).unwrap` because it's returning `Err(Msg("unexpected status code 200 OK while pushing to ..."))` instead of `Ok(_)`.

**Expected behavior**
No crash happens. 200 Response should be an acceptable response since the push was successful and I can query the metrics in prometheus.

**System information**
* CPU architecture: x86_64
* Distribution and kernel version: 5.3.5-arch1-1-ARCH
* SELinux on?: no
* Any other system details we should know?: pushgateway is running in [docker](https://hub.docker.com/layers/prom/pushgateway/latest/images/sha256-5e252ed423c397fcab03c718714464a0a8f8a0b51e075f38a38519a3eac8f171).

**Additional Info**
Not sure if this is relevant but I'm not using any authentication.

**Comment by hvariant** [comment]
from pushgateway [README](https://github.com/prometheus/pushgateway/blob/master/README.md):

```
The response code upon success is either 200 or 400. A 200 response implies a successful push, either replacing an existing group of metrics or creating a new one. A 400 response can happen if the request is malformed or if the pushed metrics are inconsistent with metrics pushed to other groups or collide with metrics of the Pushgateway itself. An explanation is returned in the body of the response and logged on error level.
```

**Comment by orfest** [comment]
Same with pushgateway v1.0

**Comment by hvariant** [comment]
> Same with pushgateway v1.0

Yeah, the official go client has a comment about this breaking change: https://github.com/prometheus/client_golang/blob/master/prometheus/push/push.go#L270

**Comment by hvariant** [comment]
I should probably close this issue since my change is merged

**Comment by sebastianjonasson** [comment]
Is this patch gonna be released anytime soon?

**Comment by rustonaut** [comment]
Run into this just now, good to know it's fixed (but not yet pushed to cargo as far as I can tell).

**Comment by pankhudinigam** [comment]
Any timelines on this fix being available to users?

**Comment by mxinden** [comment]
> Any timelines on this fix being available to users?

@pankhudinigam as far as I can tell commit https://github.com/tikv/rust-prometheus/commit/404f38ae87ac269e17b5fa635dd9ddacfa89357b is included in v0.8.0 and v0.9.0. Are you still facing the same issue with these versions?
